### ISSUE TITLE
playwright-test/(json-reporter) - add tags to suite

### ISSUE THREAD

**Issue by klh** [microsoft/playwright#24266]
https://github.com/microsoft/playwright/blob/15b9e5afdb82b2c372ce5dc0adc18540bb950d53/packages/playwright-test/src/reporters/json.ts#L156C7-L156C7

htmlreporter has added support for extracting tags from suite titles (describe("my thing @tag",{}=>())

json reporter should probably do the same.
There is also a valid point in not only extracting them,  but also removing them from the title after they have been added to a new tags: [] for the suite

**Comment by aslushnikov** [comment]
@klh HTML reporter uses tags for presentation purposes. 

JSON reporter, on the other hand, is only consumed programmatically. What exactly do you mean by "supporting tags" for JSON reporter?

**Comment by klh** [comment]
When creating a json report
The individual tests will have a tags: [], object that contains the test titles tags inside it.

That is not the case for the suite title (describe())

Which i recon it should be.

In addition tags should probably be removed fron the title post processing IF theyre moved to tags

**Comment by aslushnikov** [comment]
> That is not the case for the suite title (describe())

@klh looks like a bug then; we're open to a pull request!

> In addition tags should probably be removed fron the title post processing IF theyre moved to tags

This would be a breaking change that we can't afford.

**Comment by klh** [comment]
looking in

> > That is not the case for the suite title (describe())
> 
> @klh looks like a bug then; we're open to a pull request!
> 

looking into it

**Comment by klh** [comment]
> > That is not the case for the suite title (describe())
> 
> @klh looks like a bug then; we're open to a pull request!
> 
> > In addition tags should probably be removed fron the title post processing IF theyre moved to tags
> 
> This would be a breaking change that we can't afford.

trying to keep this PR alive, its ready to merge and builds as it should (your actions are flaky).
could you please merge this - https://github.com/microsoft/playwright/pull/24515
@aslushnikov

**Comment by klh** [comment]
Why not just merge this interim, its a tiny change and it fixes the bug.Den 15. aug. 2023 kl. 15.55 skrev Andrey Lushnikov ***@***.***>:﻿
Hey @klh,
The team decided to investigate a native tags support - tracked in #26327
So landing this PR is blocked on the results of the investigation. We'll hopefully see how it goes by 1.38

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you have additional information not present in this issue that you think will help prioritizing it, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
